src/libraries/System.Runtime.InteropServices/tests/ComInterfaceGenerator.Tests/GeneratedComClassTests.cs

```diff
@@ -0,0 +1,91 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Runtime.InteropServices;
+using System.Runtime.InteropServices.Marshalling;
+using Xunit;
+
+namespace ComInterfaceGenerator.Tests
+{
+    unsafe partial class NativeExportsNE
+    {
+        [LibraryImport(NativeExportsNE_Binary, EntryPoint = "set_com_object_data")]
+        public static partial void SetComObjectData(void* obj, int data);
+
+        [LibraryImport(NativeExportsNE_Binary, EntryPoint = "get_com_object_data")]
+        public static partial int GetComObjectData(void* obj);
+    }
+
+    [GeneratedComClass]
+    partial class ManagedObjectExposedToCom : IComInterface1
+    {
+        public int Data { get; set; }
+        int IComInterface1.GetData() => Data;
+        void IComInterface1.SetData(int n) => Data = n;
+    }
+
+    [GeneratedComClass]
+    partial class DerivedComObject : ManagedObjectExposedToCom
+    {
+    }
+
+    public unsafe class GeneratedComClassTests
```

Comment:
Is the plan to add some of the more complex cases (at least the ones in the ComObjectRedux repo) in a follow-up change?

Reply:
Yes, I'll add those in in a follow-up change.